src/delta_lake/delta_service.py

```diff
@@ -122,3 +126,43 @@ def sample_delta_table(
         raise SparkOperationError(
             f"Failed to sample rows from {full_table_name}: {str(e)}"
         )
+
+
+def query_delta_table(
+    spark: SparkSession, database: str, table: str, query: str
+) -> List[Dict[str, Any]]:
+    """
+    Executes a SQL query against a specific Delta table after basic validation.
+
+    Args:
+        spark: The SparkSession object.
+        database: The database (namespace) containing the table.
+        table: The name of the Delta table.
+        query: The SQL query string to execute.
+
+    Returns:
+        A list of dictionaries, where each dictionary represents a row.
+    """
+    _check_exists(database, table)
+    full_table_name = f"`{database}`.`{table}`"
+
+    for keyword in FORBIDDEN_KEYWORDS:
+        if keyword in query.lower():
+            raise ValueError(f"Query contains forbidden keyword: {keyword}")
+
+    # TODO: build additional validation for query to ensure it is read only
+
+    if database not in query or table not in query:
+        raise ValueError(f"Query must reference the target table [{full_table_name}]")
```

Comment:
I don't think this really adds any protection, the user could just put the table /db in comments

Reply:
no - users can get around with it easily. But better than nothing.

Reply:
I would probably remove it because it's so easy to get around and newbie SQLer's might be fooled into thinking that it actually was effective protection, but up to you

Reply:
👍